src/main/kotlin/com/icat/orboarding/user/adapters/outbound/persistence/mysql/UserPersistence.kt

```diff
@@ -34,4 +34,7 @@ class UserPersistence(private val userRepository: UserRepository) : UserPersiste
 
     private fun UserDomain.toUserEntity() =
         UserEntity(id = id!!, email = email, password = password!!)
+
+    override fun getUserByEmail(email: String): UserEntity =
```

Comment:
```suggestion
    override fun getUserByEmail(email: String): UserDomain =
```
Opa Maninho, o correto aqui é sempre externalizarmos as classes do contexto de adapter para a camada de application, pois se não, acabamos ferindo o contexto da camada de application conhecer somente as classes que são de application.